sdk/communication/azure-communication-chat/src/main/java/com/azure/communication/chat/ChatThreadAsyncClient.java

```diff
@@ -270,12 +275,28 @@ public String getChatThreadId() {
      */
     @ServiceMethod(returns = ReturnType.COLLECTION)
     public PagedFlux<ChatParticipant> listParticipants() {
+        ListParticipantsOptions listParticipantsOptions = new ListParticipantsOptions();
+        return listParticipants(listParticipantsOptions);
+    }
+
+    /**
+     * Gets the participants of a thread.
+     *
+     * @param listParticipantsOptions The request options.
+     * @return the participants of a thread.
+     */
+    @ServiceMethod(returns = ReturnType.COLLECTION)
+    public PagedFlux<ChatParticipant> listParticipants(ListParticipantsOptions listParticipantsOptions) {
         try {
             return pagedFluxConvert(new PagedFlux<>(
                 () -> withContext(context ->
-                    this.chatServiceClient.listChatParticipantsSinglePageAsync(chatThreadId, context)),
+                    this.threadClient.listChatParticipantsSinglePageAsync(
+                        chatThreadId,
+                        listParticipantsOptions.getMaxPageSize(),
```

Comment:
Here (and everywhere we do listParticipantsOptions.get...) we should have a null check on listParticipantsOptions

Check out the other methods that had pagination already, they should have null checks before invoking the threadClient with options